atomic_reactor/plugins/generate_sbom.py

```diff
@@ -181,16 +183,26 @@ def get_sbom_json(self, sbom_url: str) -> Dict[str, Any]:
                    f'but the response contains: {resp.content}.')
             raise ValueError(msg) from exc
 
-    def get_sbom_url_from_build(self, build: Dict[str, Any]) -> Optional[str]:
-        sbom_url = None
+    def get_sbom_urls_from_build(self, build: Dict[str, Any]) -> Dict[Any, str]:
+        sbom_urls = {}
 
         archives = self.koji_session.listArchives(build['build_id'], type=KOJI_BTYPE_ICM)
         sbom_path = self.pathinfo.typedir(build, btype=KOJI_BTYPE_ICM)
         for archive in archives:
-            if archive['filename'] == ICM_JSON_FILENAME:
-                sbom_url = os.path.join(sbom_path, ICM_JSON_FILENAME)
-                break
-        return sbom_url
+            for platform in self.all_platforms:
+                arch_icm_filename = ICM_JSON_FILENAME.format(platform)
```

Comment:
this nested for loop does not seem right, we should not regenerate this for every archive
I prefer something like:
```
...
icm_filenames_platforms = {ICM_JSON_FILENAME.format(platform): platform for platform in self.all_platforms}
for archive in archives:
    if archive['filename'] in icm_filenames_arches:
        platform = icm_filenames_arches[archive['filename']]
        sbom_urls[platform] = os.path.join(sbom_path, archive['filename'])
...   
```
This way we could avoid nested loop and a break statement. What do you think?

Reply:
changed